content/learn/08.contributions/03.arduino-creating-library-guide/arduino-creating-library-guide.md

```diff
@@ -125,19 +127,27 @@ First comes a couple of #include statements. These give the rest of the code acc
 #include "Morse.h"
 ```
 
-Then comes the constructor. Again, this explains what should happen when someone creates an instance of your class. In this case, the user specifies which pin they would like to use. We configure the pin as an output save it into a private variable for use in the other functions:
+Then comes the constructor. Again, this explains what should happen when someone creates an instance of your class. In this case, the user specifies which pin they would like to use. The constructor records that in a private variable for use in the other functions:
 
 ```arduino
 Morse::Morse(int pin)
 {
-  pinMode(pin, OUTPUT);
   _pin = pin;
 }
 ```
 
 There are a couple of strange things in this code. First is the **Morse::** before the name of the function. This says that the function is part of the **Morse** class. You'll see this again in the other functions in the class. The second unusual thing is the underscore in the name of our private variable, `_pin`. This variable can actually have any name you want, as long as it matches the definition in the header file. Adding an underscore to the start of the name is a common convention to make it clear which variables are private, and also to distinguish the name from that of the argument to the function (**pin** in this case).
 
-Next comes the actual code from the sketch that you're turning into a library (finally!). It looks pretty much the same, except with **Morse::** in front of the names of the functions, and `_pin` instead of **pin**:
+Next, you'll create a `begin()` function that will be called in `setup()`, after the hardware has been initialized by the Arduino main function. In our example, we need to set the pin as an output:
```

Comment:
```suggestion
Next, you'll create a `begin()` function to handle hardware configuration. This will be called from the `setup()` function of the sketch. Hardware configuration is done in a dedicated function instead of the constructor because the hardware has not yet been initialized at the time the constructor code is executed. In our library, we need to set the pin as an output:
```

- Move explanation of the reason for the `begin` function from the sketch section (accompanies my previous suggestion).
- Replace low level core information (`main`), which is not appropriate for the target reader.
- Avoid use of term "example", which might be interpreted as referring to a documentary sketch included with the library.

Reply:
Perfect! This also helps drive home the reason for the best practice of including `begin()`.